cpu/saml21/periph/pm.c

```diff
@@ -37,6 +37,17 @@ void pm_set(unsigned mode)
             case 1:
                 DEBUG_PUTS("pm_set(): setting STANDBY mode.");
                 _mode = PM_SLEEPCFG_SLEEPMODE_STANDBY;
+#ifdef MODULE_PERIPH_GPIO_IRQ
+                if (EIC->CTRLA.bit.ENABLE) {
+                    DEBUG_PUTS("pm_set(): reduce EIC clock speed");
+                    /* CKSEL is enable-protected -> disable the EIC beforehand */
+                    EIC->CTRLA.reg = 0;
+                    while (EIC->SYNCBUSY.bit.ENABLE) {}
+
+                    EIC->CTRLA.reg = EIC_CTRLA_ENABLE | EIC_CTRLA_CKSEL;
+                    while (EIC->SYNCBUSY.bit.ENABLE) {}
+                }
+#endif
```

Comment:
This code looks rather similar to the code below. Can we make it an inline function or are there already macros for enable-protected registers?

Reply:
I think it would make more sense for this to live inside `sam0_common/periph/gpio.c`, this should work the same for all sam0 other than `samd2x`

Reply:
How do you want to implement it within sam0_common/periph/gpio.c ?

Reply:
I would define a function in `cpu/sam0_common/include/periph_cpu_common.h` that is called in `cpu/caml21/periph/pm.c` and implemented in `sam0_common/periph/gpio.c`?! I think this is the solution with the least overhead.

Another solution could be some kind of registry inside of `periph/pm.c` that holds all required callbacks. `sam0_common/periph/gpio.c` would register a callback on startup. But this would imply some overhead every time `pm_set()` is called.

What solution would you prefer @benpicco?

Reply:
I had the first option in mind too.
I don't think a dynamic registry is needed here - something like what is used for auto_init would do. (Basically what you already did there)

Reply:
I was talking about a macro here to avoid code duplication as above. I guess there might be more use cases for that macro at some point.

Not sure why we switched to callbacks here?

Reply:
The code is more related to the GPIO peripheral that must be called upon PM changes. So from a logical point of view this make sense to me to put it in `periph/gpio.c`.

I already have another PR in mind: We need to mux the SPI pins back to GPIO function before switching to STANDBY oder BACKUP mode. The concept of calling certain callbacks makes sense, imho. Otherwise I have to patch this also inside of `pm.c` and clutters the whole power management.

What are your concerns regarding this callback implementation?

Reply:
That's my point. I am not talking about the callbacks in any way. I was reviewing the code *as is*. Which shows code duplication which can be solved by a nice macro and used elsewhere.

If you want to replace the above code with a callback, that's fine, but then the code will simply be added elsewhere and still be duplicated.

Reply:
Oh now I get you ... sry! Of course, I took your review serious and de-duplicated that bit. I implemented `reenable_eic(clock)`. Thank you for your review. I really appreciate it!

@benpicco and I focused on the question where to put the implementation (with or without code de-duplication). That's why we started talking about something more or less out of focus. We already had some conversation about this topic in other PRs and issues.